lib/std/math/float.zig

```diff
@@ -4,21 +4,29 @@ const expect = std.testing.expect;
 
 /// Creates a raw "1.0" mantissa for floating point type T. Used to dedupe f80 logic.
 inline fn mantissaOne(comptime T: type) comptime_int {
-    return if (@typeInfo(T).Float.bits == 80) 1 << floatFractionalBits(T) else 0;
+    return 1 << floatFractionalBits(T) & ((1 << floatMantissaBits(T)) - 1);
 }
 
 /// Creates floating point type T from an unbiased exponent and raw mantissa.
 inline fn reconstructFloat(comptime T: type, comptime exponent: comptime_int, comptime mantissa: comptime_int) T {
-    const TBits = @Type(.{ .Int = .{ .signedness = .unsigned, .bits = @bitSizeOf(T) } });
+    const FBits = @Type(.{ .Float = .{ .bits = floatBits(T) } });
+    const TBits = @Type(.{ .Int = .{ .signedness = .unsigned, .bits = floatBits(T) } });
     const biased_exponent = @as(TBits, exponent + floatExponentMax(T));
-    return @bitCast(T, (biased_exponent << floatMantissaBits(T)) | @as(TBits, mantissa));
+    return @bitCast(FBits, (biased_exponent << floatMantissaBits(T)) | @as(TBits, mantissa));
+}
+
+/// Returns the number of bits in floating point type T.
+pub inline fn floatBits(comptime T: type) comptime_int {
+    return switch (@typeInfo(T)) {
+        .Float => |info| info.bits,
+        .ComptimeFloat => 128,
```

Comment:
This shouldn't be defined in the standard library, if you need it for a behavior test then you should special case the behavior test.

Reply:
Currently this function is only used by the standard library.  I'm not clear which of the following you prefer:
1. Deleting this function and complicating every function that uses it.
2. Keeping the function, but don't export it from `math.zig`.  Instead, import `float.zig` directly when used.
3. Not adding any `comptime_float` support to these standard functions.
The only reason I exported this from `math.zig` is because it seemed helpful when trying to write generic code, as demonstrated by the other standard function changes.

Reply:
Option 3 it is then.